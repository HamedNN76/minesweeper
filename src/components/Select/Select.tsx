import React from 'react';
import {
  FormControl,
  InputLabel,
  MenuItem,
  Select as MaterialSelect,
  SelectChangeEvent,
} from '@mui/material';

type DataItem = {
  name: string;
  value: string;
};

export type SelectProps = {
  onChange: (data: SelectChangeEvent) => void;
  data: DataItem[];
  value: string;
  label: string;
};

export function Select(props: SelectProps) {
  const {data, onChange, value, label} = props;

  if (!data.length) {
    return null;
  }

  return (
    <FormControl fullWidth>
      <InputLabel id="level-select-label">{label}</InputLabel>
      <MaterialSelect
        labelId="level-select-label"
        id="level-select"
        value={value}
        label="Level"
        onChange={onChange}
      >
        {data.map((dataItem, index) => {
          return (
            <MenuItem key={index} value={dataItem.value}>
              {dataItem.name}
            </MenuItem>
          );
        })}
      </MaterialSelect>
    </FormControl>
  );
}
