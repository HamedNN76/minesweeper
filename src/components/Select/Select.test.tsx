import {cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import {Select} from './Select';
import React from 'react';
import {mount} from 'enzyme';

describe('<Select>', () => {
  afterEach(() => {
    jest.clearAllMocks();
    cleanup();
  });
  it('Match SelectBox to SnapShot', () => {
    const data = [{name: '1', value: '1'}];
    const tree = renderer
      .create(<Select data={data} value="1" label="Test" onChange={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Should return {} if data is an empty array', () => {
    const wrapper = mount(<Select data={[]} value="1" label="Test" onChange={() => {}} />);
    expect(wrapper).toEqual({});
  });
});
