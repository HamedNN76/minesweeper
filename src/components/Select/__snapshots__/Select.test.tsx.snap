// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select> Match SelectBox to SnapShot 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
>
  <label
    className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={true}
    id="level-select-label"
  >
    Test
  </label>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    onClick={[Function]}
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="level-select-label level-select"
      className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="level-select"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      1
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      disabled={false}
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value="1"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-7nxzgd"
      >
        <span>
          Level
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
